### PR TITLE
Sieve support

### PR DESCRIPTION
**Pull request by gejobj**
Dovecot, Exim4 and roundcube modifications to support sieve on Ubuntu 20.04.3 LTS

I made changes to my hestiaCP to enable sieve on it with roundcube support.
I don't know if I forgot something because I have not tried to install it from scratch, maybe I will need some help to do it.
I hope you can try it and let me know if I have to change something.
Best regards :-)

### CODE REVIEW

**Comment by jaapmarcus** [comment]
I did allready made some projects on this "feature" It also include in installer after hestia is installed. This will allow users to opt-in for it instead enabling it by default. 

https://github.com/jaapmarcus/hestiacp/commits/feature/managesieve

**Comment by gejobj** [comment]
I think it is a great feature to be added to the panel. Everyone needs to manage their emails with rules but maybe it would be nice to give the user a choice. I do not have the option to choose implemented. I made changes to my configuration files and report it as a pull request for everyone who needs it :-)

**Comment by jaapmarcus** [comment]
I don't mind adding it as a features for 1.5.0

If you need some help please join Discord as it makes communications easier..

**Comment by jlguerrerosanz** [comment]
Please move forward with this support. It is the first step to many further customizations.

**Comment by jlguerrerosanz** [comment]
I have already modified some of the dovecot conf files to add support for xz compression.

Will those changes be lost with the update?

**Comment by jaapmarcus** [comment]
In the pull request @gejobj created it will not do.

If you use mine you need first sync it to "today" but should be fine . How ever no garantuees are made...

**Comment by jaapmarcus** [comment]
Just synced my branch with the main branch. Feel free to test

**Comment by gejobj** [comment]
Hi!

I made a new shell script utility called 'v-add-sys-sieve' to install/uninstall sieve on Ubuntu 20.04.3 LTS.

I test it and it seems is working fine but, please, test and check it to find issues.
I hope this can help to make the sieve installation/remove process easier.  ^_^

I tried to include it into installation process but I don't know if I do it right beacause I can't install hestia again.

Thanks!
Best regards.

File: install/hst-install-ubuntu.sh [hunk 1]
```diff
@@ -1784,6 +1800,17 @@ if [ "$mysql" == 'yes' ] && [ "$dovecot" == "yes" ]; then
...
+#----------------------------------------------------------#
+#                     Configure Sieve                      #
+#----------------------------------------------------------#
+# Min requirements Dovecote + Exim + Mysql + roundcube
+if [ "$sieve" = 'yes' ]; then
+    echo "[ * ] Configuring Sieve ..."
+    $HESTIA/bin/v-add-sys-sieve
+else
+    # This should uninstall sieve if you had installed before
+    $HESTIA/bin/v-add-sys-sieve
```
Comment (jaapmarcus) [inline install/hst-install-ubuntu.sh]: Installer is not build for running again  so it is better to remove it...

**Review by jaapmarcus — COMMENTED** [review]
For Debian support we also need update hst-debian.sh

File: install/hst-install-ubuntu.sh (hunk 1 above)
Comment (gejobj) [reply install/hst-install-ubuntu.sh]: > nstaller is not build for running again so it is better to remove it...

I have removed that from hst-install-ubuntu.sh ;-)
Best regards!

**Comment by gejobj** [comment]
> debian

I haven't got debian OS to test v-add-sys-script and hst-debian.sh, sorry.
I need help to do that.

File: install/hst-install-ubuntu.sh [hunk 2]
```diff
@@ -182,6 +182,7 @@ for arg; do
         --postgresql)           args="${args}-g " ;;
         --exim)                 args="${args}-x " ;;
         --dovecot)              args="${args}-z " ;;
+        --sieve)                args="${args}-b " ;;
```
Comment (jaapmarcus) [inline install/hst-install-ubuntu.sh]: b is allready used by F2B replaced it with Z

File: install/hst-install-ubuntu.sh [hunk 3]
```diff
@@ -217,6 +218,7 @@ while getopts "a:w:v:j:k:m:g:d:x:z:c:t:i:b:r:o:q:l:y:s:e:p:D:fh" Option; do
         g) postgresql=$OPTARG ;;        # PostgreSQL
         x) exim=$OPTARG ;;              # Exim
         z) dovecot=$OPTARG ;;           # Dovecot
+        b) sieve=$OPTARG ;;             # Sieve
```
Comment (jaapmarcus) [inline install/hst-install-ubuntu.sh]: b is allready used by F2B replaced it with Z

**Review by jaapmarcus — COMMENTED** [review]
Made some more changes:

- Move v-add-sys-sieve to /install/upgrade/manual/ as it only should be ran once and not multiple times
- Rewrote SED command to support also for Debian (9, 10 and also 11)
- "-b" was allready in use replaced it -Z and changed default setting to no  see reason below:
- Modified default config so non sieve install will remain working

Biggest issue so far:

- After the sieve has been activated the they Quota with in Exim and Roundcube doesn't work any more. 

We should probally be able to use dovecot for limiting quotas in this case but then we need probally rewrite dovecot config a bit.

- Lack of support with in Rainloop but should  not be hard to add

**Comment by gejobj** [comment]
@jaapmarcus thank you so much for your work and help :-)

**Comment by jaapmarcus** [comment]
The old exim config had after install with the -Z flag a duplicated entry of the virtual_dilevery rule. This is also fixed.

**Review by jaapmarcus — APPROVED** [review]

**Comment by jlguerrerosanz** [comment]
Thank you very much for this addition. I know that neither eris nor scit use email. Thank you for your efforts.

**Pull request merged**